src/gadgets/mpt_update.rs

```diff
@@ -543,6 +377,293 @@ fn configure_storage<F: FieldExt>(cb: &mut ConstraintBuilder<F>, config: &MptUpd
 
 fn configure_empty_storage<F: FieldExt>(cb: &mut ConstraintBuilder<F>, config: &MptUpdateConfig) {}
 
+fn configure_poseidon_code_hash<F: FieldExt>(
+    cb: &mut ConstraintBuilder<F>,
+    config: &MptUpdateConfig,
+) {
+}
+
+fn configure_code_size<F: FieldExt>(cb: &mut ConstraintBuilder<F>, config: &MptUpdateConfig) {}
+
+fn configure_start<F: FieldExt>(cb: &mut ConstraintBuilder<F>, config: &MptUpdateConfig) {
+    cb.add_constraint(
+        "depth is 0",
+        config.selector.current(),
+        config.depth.current(),
+    );
+}
+
+fn configure_account_trie<F: FieldExt>(cb: &mut ConstraintBuilder<F>, config: &MptUpdateConfig) {
+    cb.assert(
+        "previous is Start or AccountTrie",
+        config.selector.current(),
+        config
+            .segment_type
+            .previous_matches(SegmentType::Start)
+            .or(config
+                .segment_type
+                .previous_matches(SegmentType::AccountTrie)),
+    );
+    cb.assert(
+        "next is AccountTrie or AccountLeaf0",
+        config.selector.current(),
+        config
+            .segment_type
+            .next_matches(SegmentType::AccountTrie)
+            .or(config.segment_type.matches(SegmentType::AccountLeaf0)),
+    );
+    cb.add_constraint(
+        "depth increased by 1",
+        config.selector.current(),
+        config.depth.delta() - Query::one(),
+    );
+    cb.add_constraint(
+        "0",
+        config
+            .segment_type
+            .previous_matches(SegmentType::Start)
+            .or(config
+                .segment_type
+                .previous_matches(SegmentType::AccountTrie)),
+        Query::one(),
+    );
+}
+
+fn configure_account_leaf0<F: FieldExt>(
+    cb: &mut ConstraintBuilder<F>,
+    config: &MptUpdateConfig,
+    segment_proof: &impl SegmentProofLookup,
+) {
+    cb.assert(
+        "from Start or AccountTrie",
+        config.selector.current(),
+        config
+            .segment_type
+            .previous_matches(SegmentType::Start)
+            .or(config
+                .segment_type
+                .previous_matches(SegmentType::AccountTrie)),
+    );
+    cb.assert(
+        "next is AccountLeaf1",
+        config.selector.current(),
+        config.segment_type.next_matches(SegmentType::AccountLeaf1),
+    );
+    cb.assert(
+        "path_type is Common",
+        config.selector.current(),
+        config.path_type.matches(PathType::Common),
+    );
+    cb.add_constraint(
+        "depth is 0",
+        config.selector.current(),
+        config.depth.current(),
+    );
+    cb.add_lookup(
```

Comment:
There is only one lookup for direction is required, which should be put in the top-level of `configure`

Reply:
Move segment-proof-direction lookup to `configure` function in commit [1c5d110](https://github.com/scroll-tech/mpt-circuit/pull/32/commits/1c5d110edfdb50b37015cb845a0bb1edab5481a8).